query/src/servers/http/v1/multipart_format.rs

```diff
@@ -246,6 +269,47 @@ impl Processor for SequentialInputFormatSource {
         let mut progress_values = ProgressValues::default();
         match replace(&mut self.state, State::NeedReceiveData) {
             State::ReceivedData(data) => {
+                let data = match &mut self.input_decompress {
```

Comment:
How about using constant generics?
```
SequentialInputFormatSource<const enable_decompress:bool>
```

Reply:
LGTM. And I have a plan to unify the `Format` and `Source` traits. How about refactoring them after the union? Based on my current knowledge, the `Source` trait will execute the decode logic task in async runtime?

Reply:
Ok. BTW, maybe we should not defined it as an async function if the CPU load is high